internal/pkg/argocd/argocd.go

```diff
@@ -151,17 +156,14 @@ func createArgoCdClient() (apiclient.Client, error) {
 	return clientset, nil
 }
 
-func generateDiffOfAComponent(ctx context.Context, componentPath string, prBranch string, repo string, appIf application.ApplicationServiceClient, projIf projectpkg.ProjectServiceClient, argoSettings *settings.Settings) (componentDiffResult DiffResult) {
-	componentDiffResult.ComponentPath = componentPath
-
+// This function is used to find an ArgoCD application by the SHA1 label of the component path its supposed to avoid performance issues with the "manifest-generate-paths" annotation method with requires pulling all ArgoCD applications(!) on every PR event.
```

Comment:
following [godoc convention](https://github.com/commercetools/special-delivery/blob/main/standards/CODE.md#code-comments-should-follow-godoc-conventions) for functions ([example](https://github.com/efficientgo/examples/blob/6a9ad11522ce13e274fa6557b460a83a5a4cb429/pkg/godoc/block.go#L29))

```suggestion
// findArgocdAppBySHA1Label finds an ArgoCD application by the SHA1 label of the component path it's supposed to avoid performance issues with the "manifest-generate-paths" annotation method which requires pulling all ArgoCD applications(!) on every PR event.
```